react/introduction/setting_up_a_react_environment.md

```diff
@@ -0,0 +1,145 @@
+
+The `public` folder is where all of the static assets related to your app will go. This could include images, icons, information files for the browser, and perhaps most importantly—the `index.html` file.
+
+Inside of the `src` folder is where you will find the code that runs your app. Let's open the `index.js` file and see if we can understand what's going on:
+
+~~~js
+import React from 'react';
+import ReactDOM from 'react-dom/client';
+import './index.css';
+import App from './App';
+import reportWebVitals from './reportWebVitals';
+
+const root = ReactDOM.createRoot(document.getElementById('root'));
+root.render(
+  <React.StrictMode>
+    <App />
+  </React.StrictMode>
+);
+
+// If you want to start measuring performance in your app, pass a function
+// to log results (for example: reportWebVitals(console.log))
+// or send to an analytics endpoint. Learn more: https://bit.ly/CRA-vitals
+reportWebVitals();
+~~~
+
+Whoa! There's quite a lot here. You are not expected to recognize much of this (if any) right now. Here's a brief rundown of what is happening here:
+
+1. We import `React` itself, and its fellow `ReactDOM` package.
+2. We import some CSS styling (you may recognize this syntax from the Webpack material).
+3. We import the `App` component from `App.jsx`, so that we may place (render) it within the DOM.
+4. We import the `reportWebVitals()` function, which can be used for app analytics.
+5. We create a `root` object, and assign it an element from within our `index.html`.
+6. We invoke the `render` method which is attached to our `root` object, with some very interesting looking syntax inside the parameter.
+7. We call the function `reportWebVitals()`, at this stage this can safely be ignored.
+
+All of this may understandably look unlike anything you've seen up until now, but have no fear, once you've spent the time with this course, you'll know exactly what all of this does, and _much more_.
+
+### Keeping It Clean
+
+Another of the useful features of Create React App is that it ships with [ESLint](https://www.theodinproject.com/lessons/node-path-javascript-linting). You could also set up [Prettier](https://prettier.io/) to help keep your React code as clean as can be.
+
+### Developer Tools
+
+As you progress with React, your projects will undoubtedly become larger and larger, and include more and more components, with increasing levels of functionality.
+
+When this happens, it becomes very useful to be able to track and modify aspects of your app while the page is loaded in the browser. To aid in this we can make use of a Chrome extension called [React Developer Tools](https://chrome.google.com/webstore/detail/react-developer-tools/fmkadmapgofadopljbjfkapdkoienihi?hl=en).
```

Comment:
> When this happens, it becomes very useful to be able to track and modify aspects of your app while the page is loaded in the browser.

I'm unsure about this sentence in this section. Maybe it should be phrased differently. The foremost purpose of the developer tools should be debugging rather than "modify aspects of app". I'd love to hear your thoughts on this.

Reply:
I meant more so than actually modifying the app itself, that it could be used for changing the functionality of components (state, etc.) and to see how those changes reflect in real-time, I'll make an adjustment to make this clearer and reiterate that its main purpose is debugging.

Reply:
Have made the change, please let me know if this suits.